test/forcing/NetCDFPerFeatureDataProvider_Test.cpp

```diff
@@ -0,0 +1,93 @@
+#include <vector>
+#include "gtest/gtest.h"
+#include "NetCDFPerFeatureDataProvider.hpp"
+#include "FileChecker.h"
+#include <memory>
+#include <vector>
+#include <string>
+#include <unistd.h>
+#include <stdio.h>
+#include <limits.h>
+#include <ctime>
+#include <time.h>
+
+
+using data_access::NetCDFPerFeatureDataProvider;
+
+class NetCDFPerFeatureDataProviderTest : public ::testing::Test {
+
+    protected:
+
+    void SetUp() override;
+
+    void TearDown() override;
+
+    void setupForcing();
+
+    std::shared_ptr<data_access::NetCDFPerFeatureDataProvider> nc_provider;
+
+    typedef struct tm time_type;
+
+    std::shared_ptr<time_type> start_date_time;
+
+    std::shared_ptr<time_type> end_date_time;
+
+};
+
+void NetCDFPerFeatureDataProviderTest::SetUp() {
+    //setupForcing();
+
+    setupForcing();
+}
+
+
+void NetCDFPerFeatureDataProviderTest::TearDown() 
+{
+
+}
+
+//Construct a forcing object
+void NetCDFPerFeatureDataProviderTest::setupForcing()
+{
+    nc_provider = std::make_shared<data_access::NetCDFPerFeatureDataProvider>("/local/ngen/data/huc01/huc_01/forcing/netcdf/huc01.nc");
```

Comment:
We shouldn't merge this absolute path...

Reply:
The file is to big to fit in the test data. Possibly try to make slice we can put in test data?

Reply:
Yeah, a small test data set should be suitable to check into the repo....